rfd/00TBD-webui-paginate-filter.md

```diff
@@ -0,0 +1,344 @@
+
+## Details
+
+### Pagination
+
+There is already pagination support for resources: nodes, apps, dbs. Kubes is [pending](https://github.com/gravitational/teleport/pull/9096#issuecomment-989291135)
+
+### Filter
+
+There will be filters that will be applied in order of precedence: RBAC > label > search > sort. RBAC will not be discussed since this check existed before. 
+
+### Filter: Label Query Language
+
+The query language for labels will be modeled after [kubernetes labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/). Kubernetes uses `set based` filtering which matches a label `key` against a set of `values` as shown in examples below.
+
+The language will be the same across the web ui and cli tools.
+
+Supported operators:
+- Comma between labels acts as an `AND` operator (same interpretation as current `tsh ls <label1>,<label2>`)
+- Pipe between label `values` denotes a set and acts as a `OR` operator
+- Minus before a label acts as a `NOT` operator
+
+Additionally, an empty right-hand side of the `equals` char means `any values`.
+
+| Format                        | Operator  | Example          | Description                                 |
+|-------------------------------|-----------|------------------|---------------------------------------------|
+| `key=value`                   | in        | `env=prod`       | rows with label `env=prod`                  |
+| `key=value1\|value2\|value*`  | in        | `env=prod\|dev`  | rows with labels `env=prod` or `env=dev`    |
+| `-key=value`                  | notin     | `-env=prod`      | rows without label `env=prod`               |
+| `-key=value1\|value2\|value*` | notin     | `-env=prod\|dev` | rows without labels `env=prod` or `env=dev` |
+
+<br/>
+
+| Example Combos                    | Description                                                              |
+|-----------------------------------|--------------------------------------------------------------------------|
+| `env=prod\|dev,os=mac\|linux`     | rows with labels (`env=prod` or `env=dev`) and (`os=mac` or `os=linux`)  |
+| `env=prod\|dev,os=mac,country=us` | rows with labels (`env=prod` or `env=dev`) and `os=mac` and `country=us` |
+| `-env=prod,os=mac`                | rows without label `env=prod` and with label `os=mac`                    |
+| `foo=*`                           | rows with a label with key `foo`; values unchecked                       |
+| `-foo=*`                          | rows without a label with key `foo`; values unchecked                    |
+| `foo=*,-foo=bar`                  | rows with a label with key `foo` with any values except `bar`            |
+
+<br/>
+
+In a label query, every key and its set will be translated into a list of expression objects, which will then be iterated and matched against the resource labels:
+
+```go
+type Operator int
+
+const (
+	Exists Operator = iota
+	NotExists
```

Comment:
Elsewhere in this rfd the temrs `In`/`NotInt` are used.  Either one is probably fine, but we should be consistent.